### PR TITLE
chore: add kind to devcontainer

### PR DESCRIPTION
**Pull request by olevski**
Also installs both amalthea operators in the kind cluster.

### CODE REVIEW

File: .github/workflows/save_cache.yml
Comment (leafty) [inline .github/workflows/save_cache.yml:1]: Why not use caching from [devcontainers/ci](https://github.com/devcontainers/ci)? This would cache everything we use in the devcontainer steps and is much easier to setup (add a few args to existing tasks).

Comment (olevski) [reply .github/workflows/save_cache.yml:1]: The python environment is not part of the image. So with that the python package installation will not be cached because it is not part of the image since it happens in a postCreate command after the container starts running.

And I wanted to cache the python packages.

I can add the caching of the image with devcontainer ci too.

Comment (leafty) [reply .github/workflows/save_cache.yml:1]: I think that is not exactly working the same as in a local machine. If you look at the action logs, the time spent in the `postCreate` phase is approx. ~15 seconds while the image build take 3-4 minutes. Caching the image will yield a much nicer benefit compared to doing poetry cache.

Also, it seems that poetry already caches packages in the Dockerfile.

Comment (olevski) [reply .github/workflows/save_cache.yml:1]: Ok now both the poetry cache and the devcontainer image cache should work.

Comment (olevski) [reply .github/workflows/save_cache.yml:1]: But they will come into effect once we merge this to main. I wanted to only update the caches on merges to main.

File: .github/workflows/save_cache.yml
Comment (leafty) [inline .github/workflows/save_cache.yml:31]: I think using GHCR for this would be much faster (near storage in the same infrastructure vs remote registry).

File: .github/workflows/save_cache.yml
Comment (olevski) [reply .github/workflows/save_cache.yml]: oh wow ok thanks... I was trying to understand wth is going on

File: .github/workflows/save_cache.yml
Comment (leafty) [reply .github/workflows/save_cache.yml:1]: Great!

File: .devcontainer/.poetry_cache/.gitignore
Comment (leafty) [inline .devcontainer/.poetry_cache/.gitignore:1]: non-blocking: would it also work to add `.devcontainer/.poetry_cache` to the repo `.gitignore` and force-commit a `.keep` file? This feels more conventional.

**Review by leafty — COMMENTED** [review]
The `.github` changes look good 👍.

**Comment by olevski** [comment]
@Panaetius as discussed will wait for your review here. I am already using this setup to run tests in my PR for the new operator. Currently the kind stuff is not being used.

File: .github/workflows/save_cache.yml
Comment (olevski) [reply .github/workflows/save_cache.yml:41]: This is what I wanted. As in I dont want perfect caching but as long as we can reuse the cache from main but not change it that is fine. I am a bit scared to keep updating the cache all the time.

Also my understanding of the github caches is that they cannot be used across branches. So you can only use the cache from the current branch or from a parent branch. With this in mind having this more ideal caching seemed less important because in a fresh PR you can only use the cache from main. And I worried that if we add the hash of poetry.lock then we have even less chances of hits. But we can try this.

Comment (Panaetius) [reply .github/workflows/save_cache.yml:41]: the bullet point right below leafty's in the link she posted mentions that feature branches can share the cache of the main branch, just not from one feature branch in another.

**Review by Panaetius — DISMISSED** [review]
looks good other than the comments that are already here. Let me know once those are addressed and I can reapprove

File: .github/workflows/save_cache.yml
Comment (leafty) [reply .github/workflows/save_cache.yml:41]: No, the point is that as is, you will create a single cache archive FOREVER. It will not update, which means that updates to the dependencies contained in poetry will not be cached after being merged to `main`.

Comment (leafty) [reply .github/workflows/save_cache.yml:41]: > I am a bit scared to keep updating the cache all the time.

This will not happen if the cache key reflects the `poetry.lock` contents.

Comment (olevski) [reply .github/workflows/save_cache.yml]: Ok I removed all the poetry/python caching. We can add that later.

File: .github/workflows/test_publish.yml
Comment (leafty) [inline .github/workflows/test_publish.yml:13]: non-blocking: the cache image is never created.

**Review by leafty — APPROVED** [review]

**Review by Panaetius — APPROVED** [review]

**Pull request merged**
